src/App.tsx

```diff
@@ -15,47 +16,127 @@ export const goodsFromServer = [
+    switch (sortType) {
+      case SortType.ALPABET:
+        return product1.localeCompare(product2);
+
+      case SortType.LENGTH:
+        return product1.length - product2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
 export const App: React.FC = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const goods = getReorderedGoods(goodsFromServer, { sortType, isReversed });
+
+  const handleSortListByAlphabet = () => {
+    setSortType(SortType.ALPABET);
+  };
+
+  const handleSortListByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const hendleReverse = () => {
+    setIsReversed(curent => !curent);
+  };
+
+  const handleReset = () => {
+    setIsReversed(false);
+    setSortType(SortType.NONE);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames(
+            'button is-info ',
+            { 'is-light': sortType !== SortType.ALPABET },
+          )}
+          onClick={handleSortListByAlphabet}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classNames(
+            'button is-success',
+            { 'is-light': sortType !== SortType.LENGTH },
+          )}
+          onClick={handleSortListByLength}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classNames(
+            'button is-warning',
+            { 'is-light': isReversed === false },
+          )}
+          onClick={hendleReverse}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortType !== SortType.NONE || isReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={handleReset}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {
+            goods.map(product => (
+              <li
+                data-cy="Good"
+                key={product}
+              >
+
+                {product}
+              </li>
+            ))
+          }
```

Comment:
Better create a separate component to render `GoodsList`.

Reply:
Done)